LEAF_Request_Portal/sources/FormWorkflow.php

```diff
@@ -1193,6 +1197,147 @@ public function handleEvents(int $workflowID, int $stepID, string $actionType, s
         return array('status' => 1, 'errors' => $errors);
     }
 
+    /**
+     * Get the field values of the current record
+     */
+    private function getFields(): array
+    {
+        $vars = array(':recordID' => $this->recordID);
+        $strSQL = 'SELECT `data`.`indicatorID`, `data`.`series`, `data`.`data`, `indicators`.`format`, `indicators`.`default`, `indicators`.`is_sensitive` FROM `data`
+            JOIN `indicators` USING (`indicatorID`)
+            WHERE `recordID` = :recordID';
+
+        $fields = $this->db->prepared_query($strSQL, $vars);
+        
+        $formattedFields = array();
+
+        foreach($fields as $field) 
+        {   
+            if ($field["is_sensitive"] == 1) {
+                $formattedFields[$field['indicatorID']] = "**********";
+                continue;
+            }
+
+            $format = $field["format"];
+            $data = $field["data"];
+
+            switch(true) {

```

Comment:
Ok.  Need to make these case insensitive.  Recommend strToLower() on $format above.